File: javascript/K번째수.js
const solution = (array, commands) => {
    let answer = [];
    for (let i = 0; i < commands.length; i++){
        let selectedArray = array.slice(commands[i][0] - 1, commands[i][1]);
        selectedArray.sort((a,b)=>a-b);
        answer.push(selectedArray[commands[i][2] - 1]);
    }
    return answer;
}

File: java/K번째수.java
import java.util.*;

class Solution {
    public static int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        int index = 0 ;

        for(int i = 0 ; i < commands.length ; i++)
        {
            int start = commands[i][0];
            int end = commands[i][1];
            int k = commands[i][2];
            int[] tmp = new int[end - start + 1];
            int a = 0;
            
            for(int j = start - 1 ; j < end ; j++) {
                tmp[a++] = array[j];
            }
            
            Arrays.sort(tmp);
            answer[index++] = tmp[k-1];
        }
        return answer;
    }
}